### PR TITLE
Migrate to new LEP engine

### PR DESCRIPTION
**Pull request by d-bagazhkov**

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepContextFactory.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    private final MeterRegistry meterRegistry;
+
+    @Override
+    public BaseLepContext buildLepContext(LepMethod lepMethod) {
+        LepContext lepContext = new LepContext();
+        lepContext.services = new LepContext.LepServices();
+        lepContext.services.tenantConfigService = tenantConfigService;
+        lepContext.services.permissionService = permissionCheckService;
+        lepContext.services.mailService = mailService;
+        lepContext.services.commonsService = new CommonsExecutor(commonsService);
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepContextFactory.java:37]: looks like mistake

File: src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java [hunk 2]
```diff
@@ -38,29 +22,25 @@ public class XmCommunicationMsLepProcessingApplicationListenerTest {
...
     @Test
     @SuppressWarnings("unchecked")
     public void testBindExecutionContext() {
 
-        ScopedContext context = new DefaultScopedContext("scope");
-        listener.bindExecutionContext(context);
+        LepContext context = (LepContext) contextFactory.buildLepContext(null);
 
-        assertEquals(4, context.getValues().size());
-        assertNotNull(context.getValue(BINDING_KEY_COMMONS, CommonsExecutor.class));
```
Comment (sergeysenja1992) [inline src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java:51]: pls annotate LepContext by GroovyMap annotation, and just replace getValue to get method, and test have to work without additional changes

this test must work without changes, fail of this test indicate potential fail of existing lep

File: src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java [hunk 3]
```diff
@@ -38,29 +22,25 @@ public class XmCommunicationMsLepProcessingApplicationListenerTest {
...
-        assertNotNull(context.getValue(BINDING_KEY_SERVICES, Map.class));
```
Comment (sergeysenja1992) [inline src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java:52]: this check must be changed from map to class

File: src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java [hunk 4]
```diff
@@ -38,29 +22,25 @@ public class XmCommunicationMsLepProcessingApplicationListenerTest {
...
-        assertNotNull(context.getValue(BINDING_KEY_TEMPLATES, Map.class));
```
Comment (sergeysenja1992) [inline src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java:53]: and this

File: src/main/java/com/icthh/xm/tmf/ms/communication/lep/LepContext.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    public LepServices services;
+    public LepTemplates templates;
+
+    public static class LepServices {
+        public MailService mailService;
+        public MeterRegistry meterRegistry;
+        public BusinessTimeConfigService tenantConfigService;
+        public PermissionCheckService permissionService;
+        public CommonsExecutor commonsService;
```
[the remark this answers is not in the document]
Comment (d-bagazhkov) [reply src/main/java/com/icthh/xm/tmf/ms/communication/lep/LepContext.java]: fixed

File: src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepContextFactory.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        lepContext.services.meterRegistry = meterRegistry;
```
[the remark this answers is not in the document]
Comment (d-bagazhkov) [reply src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepContextFactory.java]: fixed

File: src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java (hunk 2 above)
Comment (d-bagazhkov) [reply src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java:51]: I added a second test

File: src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java (hunk 3 above)
Comment (d-bagazhkov) [reply src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java:52]: I added a second test

File: src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java (hunk 4 above)
Comment (d-bagazhkov) [reply src/test/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListenerTest.java:53]: I added a second test

File: src/main/java/com/icthh/xm/tmf/ms/communication/lep/LepContext.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import com.icthh.xm.tmf.ms.communication.service.mail.MailService;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.springframework.web.client.RestTemplate;
+
+@GroovyMap
+public class LepContext extends BaseLepContext  {
+
+    public LepServices services;
+    public LepTemplates templates;
+    public CommonsExecutor commonsService;
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/lep/LepContext.java:18]: looks like it already present in BaseLepContext ?

File: src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepContextFactory.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    private final PermissionCheckService permissionCheckService;
+    private final KafkaTemplateService kafkaTemplateService;
+    private final MailService mailService;
+    private final MeterRegistry meterRegistry;
+
+    @Override
+    public BaseLepContext buildLepContext(LepMethod lepMethod) {
+        LepContext lepContext = new LepContext();
+
+        lepContext.commonsService = new CommonsExecutor(commonsService);
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepContextFactory.java:34]: looks like it already present in BaseLepContext ?

**Review by sergeysenja1992 — APPROVED** [review]

**Review by major13ua — APPROVED** [review]

**Pull request merged**